conda_auth/cli.py

```diff
@@ -83,18 +76,58 @@ def auth_wrapper(args):
 
 
 @group.command("login")
-@click.argument("channel", callback=validate_channel)
-def login(channel: ChannelData):
+@click.option("-u", "--username", help="Username to use for HTTP Basic Authentication")
+@click.option("-p", "--password", help="Password to use for HTTP Basic Authentication")
+@click.option(
+    "-t",
+    "--type",
+    help='Manually specify the type of authentication to use. Choices are: "http-basic"',
+)
+@click.argument("channel", callback=parse_channel)
+def login(channel: Channel, **kwargs):
     """
     Login to a channel
     """
-    channel.manager.authenticate(channel.channel, channel.settings)
+    kwargs = {key: val for key, val in kwargs.items() if val is not None}
+    settings = get_channel_settings(channel.canonical_name) or {}
+    settings.update(kwargs)
+
+    auth_type, auth_manager = get_auth_manager(settings)
+    attempts = 0
+
+    while True:
+        try:
+            username = auth_manager.authenticate(channel, settings)
+            break
+        except InvalidCredentialsError as exc:
+            auth_manager.remove_channel_cache(channel.canonical_name)
+            attempts += 1
+            if attempts >= MAX_LOGIN_ATTEMPTS:
+                raise CondaAuthError(f"Max attempts reached; {exc}")
+
+    click.echo(click.style(SUCCESSFUL_LOGIN_MESSAGE, fg=SUCCESSFUL_COLOR))
+
+    try:
+        condarc = CondaRC()
+        condarc.update_channel_settings(channel.canonical_name, username, auth_type)
+        condarc.save()
+    except CondaRCError as exc:
+        raise CondaAuthError(str(exc))
 
 
 @group.command("logout")
-@click.argument("channel", callback=validate_channel)
-def logout(channel):
+@click.argument("channel", callback=parse_channel)
```

Comment:
Do channels ever have multiple/separate users? In that case, having a `--username` option here would be good (but I'm not sure if this is something that is necessary)

Reply:
That's a fair question, and I thought about this too. We could support this, but I don't know how many people are actually doing this.

I would just leave it open as a possibility for something to implement in the future if requested.